src/Microsoft.Management.Configuration.UnitTests/Tests/ConfigurationMixedElevationTests.cs

```diff
@@ -0,0 +1,176 @@
+                Assert.Equal(ConfigurationUnitResultSource.None, unitResult.ResultInformation.ResultSource);
+            }
+        }
+
+        /// <summary>
+        /// Verifies the behavior of running a restricted unit.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task ApplyRestrictedUnit()
+        {
+            string resourceName = "E2ETestResource";
+            string moduleName = "xE2ETestResource";
+            Version version = new Version("0.0.0.1");
+
+            ConfigurationSet configurationSet = this.ConfigurationSet();
+            configurationSet.Metadata.Add(Helpers.Constants.EnableDynamicFactoryTestMode, true);
+            configurationSet.Metadata.Add(Helpers.Constants.EnableRestrictedIntegrityLevelTestGuid, true);
+
+            ConfigurationUnit elevationRequiredUnit = this.ConfigurationUnit();
+            elevationRequiredUnit.Metadata.Add("securityContext", "restricted");
+            elevationRequiredUnit.Metadata.Add("version", version.ToString());
+            elevationRequiredUnit.Metadata.Add("module", moduleName);
+            elevationRequiredUnit.Type = resourceName;
+            elevationRequiredUnit.Intent = ConfigurationUnitIntent.Apply;
+
+            ConfigurationUnit unit = this.ConfigurationUnit();
+            unit.Metadata.Add("version", version.ToString());
+            unit.Metadata.Add("module", moduleName);
+            unit.Type = resourceName;
+            unit.Intent = ConfigurationUnitIntent.Apply;
+
+            configurationSet.Units = new ConfigurationUnit[] { elevationRequiredUnit, unit };
+
+            IConfigurationSetProcessorFactory dynamicFactory = await this.fixture.ConfigurationStatics.CreateConfigurationSetProcessorFactoryAsync(Helpers.Constants.DynamicRuntimeHandlerIdentifier);
+
+            ConfigurationProcessor processor = this.CreateConfigurationProcessorWithDiagnostics(dynamicFactory);
+
+            ApplyConfigurationSetResult result = processor.ApplySet(configurationSet, ApplyConfigurationSetFlags.None);
+            Assert.NotNull(result);
+            Assert.Null(result.ResultCode);
+            Assert.Equal(2, result.UnitResults.Count);
+
+            foreach (var unitResult in result.UnitResults)
+            {
+                Assert.NotNull(unitResult);
+                Assert.False(unitResult.PreviouslyInDesiredState);
+                Assert.False(unitResult.RebootRequired);
+                Assert.NotNull(unitResult.ResultInformation);
+                Assert.Null(unitResult.ResultInformation.ResultCode);
+                Assert.Equal(ConfigurationUnitResultSource.None, unitResult.ResultInformation.ResultSource);
+            }
+        }
+
+        /// <summary>
+        /// Verifies that creating a high integrity unit processor for a non elevated unit should return an invalid operation result.

```

Comment:
This test should have an elevated unit as well so that we can see one passing and one failing.